### PR TITLE
release-23.1: upgrade: remove buggy TTL repair

### LINKED ISSUE #110363: upgrade: automated catalog repairs incorrectly removes TTL from the table descriptor
During a cluster upgrade TTL is incorrectly removed from table descriptors [here](https://github.com/cockroachdb/cockroach/blob/v23.1.9/pkg/sql/catalog/tabledesc/table_desc_builder.go#L250) because the TTL job schedule ID from the table descriptor is incorrectly being searched for in a set of job IDs [here](https://github.com/cockroachdb/cockroach/blob/v23.1.9/pkg/sql/sem/builtins/builtins.go#L5216).

Later on when the TTL job runs, it panics (stack trace here https://github.com/cockroachlabs/support/issues/2593#issuecomment-1713899436) and crashes the node.

**Remove the TTL "repair" for now:** https://cockroachlabs.slack.com/archives/C05RE28U495/p1694456292147979?thread_ts=1694454150.432349&cid=C05RE28U495

See also:
https://github.com/cockroachlabs/support/issues/2593#issuecomment-1714377621

Jira issue: CRDB-31400

### PR DESCRIPTION
**Pull request by ecwall**
Backport 2/2 commits from #110364 and https://github.com/cockroachdb/cockroach/pull/110816.

/cc @cockroachdb/release

---

Fixes #110363

The TTL descriptor repair in FirstUpgradeFromReleasePrecondition incorrectly
removes TTL fields from table descriptors after incorrectly comparing the
table descriptor's TTL job schedule ID to a set of job IDs.

This change removes the repair until tests are properly added.

Release note (bug fix): Remove buggy TTL descriptor repair. Previously,
upgrading from 22.2.X to 23.1.9 incorrectly removed TTL storage params from
tables (visible via `SHOW CREATE TABLE <ttl-table>;`) while attempting to
repair table descriptors. This resulted in the node that attempts to run the
TTL job crashing due to a panic caused by the missing TTL storage params.
Clusters currently on 22.2.X should NOT be upgraded to 23.1.9 and should
be upgraded to 23.1.10 or later directly.

Release justification: Backport to base extraordinary release 23.1.10 on.

### CODE REVIEW

**Review by rafiss — CHANGES_REQUESTED** [review]
nit: let's add an additional change to this backport. it should use an environment variable to make this function return early by default: https://github.com/cockroachdb/cockroach/blob/f2b6dc7f4cfa4542054d864a09f83a4f3045e0b7/pkg/upgrade/upgrades/first_upgrade.go#L38

File: pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+statement ok
+CREATE TABLE tbl (
+  id INT PRIMARY KEY
+) WITH (ttl_expire_after = '10 minutes')
+
+upgrade all
+
+query B retry
+SELECT version LIKE '%23.1-%' FROM [SHOW CLUSTER SETTING version]
```
Comment (rafiss) [inline pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors:11]: in the backport, this should be `LIKE '%22.2-%'`

**Review by ecwall — COMMENTED** [review]
Added

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110500#-:-NeBTZUtBIFYMG-7014p:b-8b11j9)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors` line 11 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/110500#gh-1323554034:-NeBTWXN8jAHCzitzCMa:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/daf99dead7257dcad36e2e85d5fd5168289b8268/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors#L11)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

in the backport, this should be `LIKE '%22.2-%'`
</blockquote></details>

Done

**Review by rafiss — APPROVED** [review]
nice, lgtm!

**Comment by rafiss** [comment]
blathers backport 23.1.11-rc

**Comment by celiala** [comment]
blathers backport staging-v23.1.10

**Pull request merged**